internal/service/finspace/kx_cluster_test.go

```diff
@@ -269,6 +337,140 @@ func TestAccFinSpaceKxCluster_rdb(t *testing.T) {
 	})
 }
 
+func TestAccFinSpaceKxCluster_rdbInvalidEnvironmentId(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping long-running test in short mode")
+	}
+
+	ctx := acctest.Context(t)
+	var kxEnvironment finspace.GetKxEnvironmentOutput
+	rName := sdkacctest.RandomWithPrefix(acctest.ResourcePrefix)
+	invalidEnvironmentId := sdkacctest.RandString(22)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck: func() {
+			acctest.PreCheck(ctx, t)
+			acctest.PreCheckPartitionHasService(t, finspace.ServiceID)
+		},
+		ErrorCheck:               acctest.ErrorCheck(t, finspace.ServiceID),
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories,
+		CheckDestroy:             testAccCheckKxClusterDestroy(ctx),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccKxClusterConfigBase(rName),
+				Check:  testAccCheckKxEnvironmentExists(ctx, "aws_finspace_kx_environment.test", &kxEnvironment),
+			},
+			{
+				Config:      testAccKxClusterConfig_rdbEnvironmentId(rName, invalidEnvironmentId),
+				ExpectError: regexp.MustCompile("ResourceNotFoundException: No matching Environment found with requested ID"),
+			},
+		},
+	})
+}
+
+func TestAccFinSpaceKxCluster_rdbInvalidDatabaseName(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping long-running test in short mode")
+	}
+
+	ctx := acctest.Context(t)
+	var kxEnvironment finspace.GetKxEnvironmentOutput
+	rName := sdkacctest.RandomWithPrefix(acctest.ResourcePrefix)
+	databaseName := "invalid-db"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck: func() {
+			acctest.PreCheck(ctx, t)
+			acctest.PreCheckPartitionHasService(t, finspace.ServiceID)
+		},
+		ErrorCheck:               acctest.ErrorCheck(t, finspace.ServiceID),
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories,
+		CheckDestroy:             testAccCheckKxClusterDestroy(ctx),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccKxClusterConfigBase(rName),
```

Comment:
Shouldn't need this as a separate test step I think, but if we want to confirm environment is created before we test the failure case that should be ok

Reply:
Is this case any different from the invalid environment id for the hdb and rdb clusters? For all these tests I wanted to confirm that the environment was created correctly before negative tests

Reply:
That makes sense. In that case we can leave it as is